tool/src/org/antlr/v4/codegen/target/GoTarget.java

```diff
@@ -134,7 +134,7 @@ public String encodeIntAsCharEscape(int v) {
 
 	@Override
 	public int getSerializedATNSegmentLimit() {
-		return 2 ^ 31;
+		return 29;
```

Comment:
Constant folding does not improve code readability. Moreover, compiler does it itself.

Also, I'm not sure it's the correct value here. There are much greater values for other runtimes: [Python3Target](https://github.com/antlr/antlr4/blob/7b32c9b169a2d9dd2375d4fc9c5e678f0ed0039c/tool/src/org/antlr/v4/codegen/target/Python3Target.java#L60), [SwiftTarget.java](https://github.com/antlr/antlr4/blob/7b32c9b169a2d9dd2375d4fc9c5e678f0ed0039c/tool/src/org/antlr/v4/codegen/target/SwiftTarget.java#L111), and [others](https://github.com/antlr/antlr4/search?q=getSerializedATNSegmentLimit).

Reply:
Correct. The problem is that `2 ^ 31` is XOR and not power. The static analyzer identified that it is confusing as written as it always evaluates to 29 and is likely incorrect. I simply changed it so it is clear. Since most other targets use `65535 / 3`, I've switched it to that.

Reply:
for the life of me I can't remember why we are doing 2^31...XOR? weird.  Oh right, yeah, let's use:
`return 65535 / 3` like Java.

Reply:
I think it makes sense to fix this value for the [JavaScript](https://github.com/antlr/antlr4/blob/7b32c9b169a2d9dd2375d4fc9c5e678f0ed0039c/tool/src/org/antlr/v4/codegen/target/JavaScriptTarget.java#L91) target. It's also incorrect.